### PR TITLE
#BN-25 feat: Create and view available products from vendor's collection

### PR DESCRIPTION
**Pull request by EvelyneUmubyeyi**
# Trello Card Link
https://trello.com/c/ZNGWRaaA

# PR Description
This PR contains an endpoint for toggling product availability status between true and false for vendors. A vendor might need to change product availability status due to running out of stock, expired stock, etc. 

It also contains an endpoint for retrieving all available products from a vendor's collection and another one for retrieving unavailable products.

# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] Create an endpoint for changing product availability status for vendors
- [x] Responding with appropriate error messages in case of unsuccessful attempts
- [x] Retrieving available products from the vendor's collection
- [x] Retrieving unavailable products from the vendor's collection
- [x] Testing and documenting endpoints created

# Description of tasks that were expected to be completed and that are not yet completed
- [x] None

# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run test`
3. Also try the endpoint in [Swagger](http://localhost:9090/api-docs/#/Products%20Availability/post_product_availability)

# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

**Review by kunda4 — APPROVED** [review]
Well done

**Review by Niyobuzima — APPROVED** [review]
**Well Done**

**Review by Niyobuzima — DISMISSED** [review]
**Well Done**

**Review by kunda4 — DISMISSED** [review]

**Review by Charlesntwari — DISMISSED** [review]

**Review by Lydie260 — DISMISSED** [review]

**Review by Charlesntwari — APPROVED** [review]

File: .env.example
Comment (kunda4) [inline .env.example:1]: This file doesn't have to be here

**Review by kunda4 — APPROVED** [review]

**Review by kunda4 — DISMISSED** [review]

**Review by Charlesntwari — APPROVED** [review]

**Review by Charlesntwari — DISMISSED** [review]

**Review by Niyobuzima — DISMISSED** [review]
**good job**

File: src/controllers/vendor/vendorsController.js [hunk 1]
```diff
@@ -70,32 +70,46 @@ export const registerVendor = async (req, res) => {
...
+        where: {
+          id: req.body.userId,
+        },
+        returning: true,
+      },
+    );
+
+    const recipientEmail = User.dataValues.email;
     const emailSubject = 'Welcome to Cogito Ecommerce';
-    const emailMessage = `Dear ${req.body.fullName},\n\nWelcome to Cogito! We are excited to have you as a new vendor. Your business information has been successfully saved in our system.\n\nHere is the information you provided:\n\nFull Name: ${req.body.fullName}\nBusiness Name: ${req.body.businessName}\nBusiness Address: ${req.body.businessAddress}\nBusiness Phone Number: ${req.body.businessPhoneNumber}\nBusiness Email: ${req.body.businessEmail}\nBusiness Website: ${req.body.businessWebsite}\nBusiness Description: ${req.body.businessDescription}\nProduct Categories: ${req.body.productCategories}\nPayment Methods: ${req.body.paymentMethods}\nRandom Password: ${req.body.password}, You can change this password only after login to our site.\n Login Here:\t http://localhost:9090/vendors/login \n\nThank you for choosing Cogito!\n\nBest regards,\nThe Cogito Team`;
```
Comment (dushimeemma) [inline src/controllers/vendor/vendorsController.js:85]: Would you mind to use a helper function for the email?

Comment (EvelyneUmubyeyi) [reply src/controllers/vendor/vendorsController.js:85]: Done!

**Review by Niyobuzima — DISMISSED** [review]

**Review by Charlesntwari — APPROVED** [review]

**Review by Lydie260 — APPROVED** [review]

**Review by Lydie260 — APPROVED** [review]

**Review by dushimeemma — APPROVED** [review]
Good job 🥇

**Pull request merged**